### ISSUE TITLE
os_server: Error reporting with floating IP yet resources are still created

### ISSUE THREAD

**Issue by Thingee** [ansible/ansible#19613]
_From @lowwalker on August 28, 2015 11:41_

##### Issue Type:
- Bug Report
##### Ansible Version:

```
➜  luket-testing  ansible --version
ansible 2.0.0 (devel 5ba3452b7e) last updated 2015/08/27 09:58:27 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 85ddb1b902) last updated 2015/08/25 08:53:22 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 7a0bfd91bb) last updated 2015/08/25 08:53:23 (GMT -400)
  v1/ansible/modules/core: (detached HEAD f8d8af17cd) last updated 2015/08/10 09:31:46 (GMT -400)
  v1/ansible/modules/extras: (detached HEAD 495ad450e5) last updated 2015/08/10 09:31:53 (GMT -400)
  config file =
  configured module search path = None
```
##### Ansible Configuration:

Whatever is provided by "source ./hacking/env-setup"
##### Environment:

OSX 10.10.4 for the control machine, if needed I'll track down the specific Openstack version,
##### Summary:

If there is no configured floating-ip networks the auto_floating_ip option set to true causes the below error, however the instance is still created. 

```
➜  luket-testing  ap -i hosts ostack_inital.yml

PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [127.0.0.1]

TASK [Launch instance] *********************************************************
fatal: [127.0.0.1]: FAILED! => {"changed": false, "extra_data": null, "failed": true, "msg": "unable to bind a floating ip to server 52f5b0fb-b18f-44ad-8d2b-f50b7d555829: External network 8f3508a9-d4f5-4f9c-a5da-fd7f04059303 is not reachable from subnet 6734ec38-0611-49ed-8007-9fdb1631befc.  Therefore, cannot associate Port f5ae8af1-6871-4ea2-b9fa-8ba10c67885d with a Floating IP."}

PLAY RECAP *********************************************************************
127.0.0.1                  : ok=1    changed=0    unreachable=0    failed=1
```
##### Steps To Reproduce:

Simple playbook to launch instance.

```
    - name: Launch instance
      os_server:
        state: present
        auth:
          auth_url: https://somereallycoolurl.com:5678/v2.0
          username: ltislow
          password: FixItNowBENNO!
          project_name: initial-testing
        name: luke-console-ansible
        region_name: region-3
        image: 3b0673fb-8ca5-4a52-b5f5-c728e709cd9d
        key_name: Luke Tislow Laptop
        timeout: 300
        flavor: Medium
        security_groups: default
        auto_floating_ip: yes
```
##### Expected Results:

When failure from network configuration options, do not create the instance. Or improve error message that instance was created yet not able to complete the network configuration.
##### Actual Results:

Instance is still created, if the playbook is ran again the same error message is provided and no further instances are created.

_Copied from original issue: ansible/ansible-modules-core#1981_

**Comment by Thingee** [comment]
_From @abadger on September 1, 2015 22:22_

/cc @emonty

**Comment by Thingee** [comment]
_From @emonty on September 2, 2015 16:9_

Oh yuck. Thanks - will work up a fix.

**Comment by Thingee** [comment]
_From @gregdek on September 15, 2015 20:0_

Also cc: @shrews @dguerri @juliakreger @j2sol @rcarrillocruz 

Just in case :)

**Comment by Thingee** [comment]
_From @ansibot on August 1, 2016 2:48_

@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by Thingee** [comment]
_From @ansibot on September 12, 2016 21:1_

@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Thingee** [comment]
_From @ansibot on October 6, 2016 15:50_

@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Thingee** [comment]
_From @ansibot on October 22, 2016 9:19_

@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Thingee** [comment]
_From @ansibot on November 6, 2016 10:9_

@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Thingee** [comment]
_From @ansibot on November 22, 2016 10:17_

@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz, @thingee, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Thingee** [comment]
_From @herlo on December 1, 2016 19:16_

https://review.openstack.org/#/c/393884 This review is merged and supposedly addresses the floating_ip issue. @emonty

**Comment by Thingee** [comment]
_From @ansibot on December 7, 2016 18:14_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
@Thingee Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibotdev** [comment]
@Thingee You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
